src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,218 @@
+from pathlib import Path
+
+from ophyd_async.panda import HDFPanda
+
+from dodal.common.beamlines.beamline_utils import (
+    device_instantiation,
+    get_directory_provider,
+    set_directory_provider,
+)
+from dodal.common.beamlines.beamline_utils import set_beamline as set_utils_beamline
+from dodal.common.beamlines.device_helpers import numbered_slits
+from dodal.common.visit import DirectoryServiceClient, StaticVisitDirectoryProvider
+from dodal.devices.focusing_mirror import FocusingMirror
+from dodal.devices.i22.dcm import CrystalMetadata, DoubleCrystalMonochromator
+from dodal.devices.slits import Slits
+from dodal.devices.synchrotron import Synchrotron
+from dodal.devices.tetramm import TetrammDetector
+from dodal.devices.undulator import Undulator
+from dodal.devices.xspress3.xspress3 import Xspress3
+from dodal.log import set_beamline as set_log_beamline
+from dodal.utils import BeamlinePrefix, get_beamline_name, skip_device
+
+BL = get_beamline_name("i18")
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+#
+# table again, t1 theta and d7bdiode
+# MO table is ok
+#
+# pinhole PVs?
+# -AL-APTR-01:TEMP1
+#
+# diode reading and also DRAIN
+# diode is ok, there are A and B variants
+# motors A and B
+# camera not used
+
+# Currently we must hard-code the visit, determining the visit at runtime requires
+# infrastructure that is still WIP.
+# Communication with GDA is also WIP so for now we determine an arbitrary scan number
+# locally and write the commissioning directory. The scan number is not guaranteed to
+# be unique and the data is at risk - this configuration is for testing only.
```

Comment:
Should: How do these comments help the reader understand the code? The first set are in such note form that they're hard to understand and the second set would be better written as a GH issue on what needs to be done precisely and the comment just link to that issue

Reply:
deleted